db/seeds.rb

```diff
@@ -234,101 +192,23 @@ def generate_checkout
   end
 end
 
-def generate_blackout
-  Blackout.create! do |blk|
-    blk.start_date = time_rand(Time.zone.now + 1.year)
-    # rubocop:disable Date
-    blk.end_date = time_rand(blk.start_date.to_time,
-                             blk.start_date.next_week.to_time)
-    # rubocop:enable Date
-    blk.notice = FFaker::HipsterIpsum.paragraph(2)
-    blk.created_by = User.first.id
-    blk.blackout_type = %w(soft hard).sample
-  end
-end
-
-def mark_checked_in(res, checkout_length)
-  return unless res.checked_out
-  if rand < OVERDUE_CHANCE
-    res.overdue = true
-    res.status = 'checked_out'
-    res.checked_in = nil
-  else
-    res.status = 'returned'
-    res.checked_in = time_rand(res.checked_out, res.checked_out.next_week,
-                               checkout_length).to_datetime
-    res.checkin_handler_id = User.where('role = ? OR role = ? OR role = ?',
-                                        'checkout', 'admin', 'superuser'
-                                       ).all.sample.id
-  end
-end
-
-def mark_checked_out(res)
-  if rand < MISSED_CHANCE
-    res.checked_out = nil
-    res.status = 'missed'
-  else
-    res.status = 'checked_out'
-    res.checked_out = res.start_date
-    res.equipment_item = res.equipment_model.equipment_items.all.sample
-    res.checkout_handler_id = User.where('role = ? OR role = ? OR role = ?',
-                                         'checkout', 'admin', 'superuser'
-                                        ).all.sample.id
-  end
-end
-
-def throw_into_past(res)
-  factor = rand(1.day..PAST_RANGE)
-  res.start_date = res.start_date - factor
-  res.due_date = res.due_date - factor
-  travel_to(res.start_date) { res.save }
-  mark_checked_out res
-  return if res.status == 'missed' || rand < RETURNED_CHANCE
-
-  mark_checked_in(res, res.equipment_model.category.max_checkout_length)
-  travel_to(res.due_date) { res.save } unless res.overdue
+def generate_res
+  ReservationGenerator.generate_random
 end
 
-# rubocop:disable AbcSize, MethodLength
-def generate_reservation
-  count = 0
-  while count < MAX_TRIES
-    res = Reservation.new
-    res.status = 'reserved'
-    res.reserver_id = User.all.sample.id
-    res.equipment_model = EquipmentModel.all.sample
-    res.start_date = time_rand(Time.zone.now,
```

Comment:
So now that you've stopped using `time_rand` with reservation generation, we're just relying on our validations to reject any reservations that overlap with blackout dates? That actually seems much more elegant to me, I'm wondering if we even need to include that functionality for Blackouts... do we care if they overlap? I'll comment more in the module